File: Creacionales/Typescript/Singleton Pattern/ConexionDb.ts
export class ConexionDb{
    
    public static instancia:ConexionDb;
    
    consturctor(){
        console.log("Conexión establecida");
    }

    static getInstancia():ConexionDb{
        if(ConexionDb.instancia==null){
            ConexionDb.instancia=new ConexionDb();
        }
        return ConexionDb.instancia;
    }

    ejecutarQuery(sql:string):void{
        console.log("Ejecutando query: "+sql);
    }
    
}

File: Creacionales/Java/Singleton/src/ConexionDB.java
public class ConexionDB {

    private static ConexionDB instance;

    private ConexionDB(){}

    public static ConexionDB getConexion(){
        if(instance==null){
            instance = new ConexionDB();
        }
        return instance;
    }

    public void exexcuteQery(String query){
        System.out.println(query);
    }
}
